ChildProject/pipelines/eafbuilder.py

```diff
@@ -1,28 +1,120 @@
+    eaf.add_tier("context_"+eaf_type, ling=ling_type)
+    eaf.add_tier("code_num_"+eaf_type, ling=ling_type)
+
+    for i, ts in enumerate(timestamps_list):
+        print("Creating eaf code segment # ", i+1)
+        print("enumerate makes: ", i, ts)
+        whole_region_onset = ts[0]
+        whole_region_offset = ts[1]
+
+        context_onset = int(whole_region_onset) - contxt_on
+        context_offset = int(whole_region_offset) + contxt_off
+
+        if context_onset < 0:
+            context_onset = 0.0
+        
+        codeNumVal = eaf_type + str(i+1)
+        eaf.add_annotation("code_"+eaf_type, whole_region_onset, whole_region_offset)
+        eaf.add_annotation("code_num_"+eaf_type, whole_region_onset, whole_region_offset, value=codeNumVal)
+        eaf.add_annotation("context_"+eaf_type, context_onset, context_offset)
+
+    os.makedirs(output_dir, exist_ok = True)
+    eaf.to_file(os.path.join(output_dir, "{}.eaf".format(id)))
+    for i in eaf.get_tier_names():
+        print(i,":",eaf.get_annotation_data_for_tier(i))
+
+    return eaf
+
 class EafBuilderPipeline(Pipeline):
     def __init__(self):
         pass
                 
-    def run(self, path: str, destination: str, segments: str,
-        type: str, template: str,
-        context_onset: float, context_offset: float,
+    def run(self, destination: str, segments: str,
+        eaf_type: str, template: str,
+        context_onset: int = 0, context_offset: int = 0,
         **kwargs):
+        """generate .eaf templates based on intervals to code
+
+        :param path: project path
+        :type path: str
+        :param destination: eaf destination
+        :type destination: str
+        :param segments: path to the input segments dataframe
+        :type segments: str
+        :param eaf_type: eaf-type [random, periodic]
+        :type eaf_type: str
+        :param template: name of the template to use (basic, native, or non-native)
+        :type template: str
+        :param context_onset: context onset and segment offset difference in milliseconds, 0 for no introductory context
+        :type context_onset: float
+        :param context_offset: context offset and segment offset difference in milliseconds, 0 for no outro context
+        :type context_offset: float
+        """
+
+        # TODO: Make sure etf file paths are approprite and robust. 
+        etf_path = "{}.etf".format(template)
+        psfx_path = "{}.pfsx".format(template)
+
+        print("making the "+eaf_type+" eaf file and csv")
+
+        segments = pd.read_csv(segments)
+
+        for recording_filename, segs in segments.groupby('recording_filename'):
+            recording_filename = os.path.splitext(recording_filename)[0]
+            output_filename = recording_filename + '_' + eaf_type + '_' + template
+            
+            # TODO: This list of timestamps as tuples might not be ideal/should perhaps be optimized, but I am just replicating the original eaf creation code here.
```

Comment:
what is the alternative -- is it a csv file with on/off?

Reply:
Yeah, essentially that is what the eaf file comes to be. Just a bunch of tiers with empty annotations with the segments. So we can output csv too, but that is already output by the samplers. Though I placed that TODO more in order to double check if this is the efficient way to pull out segments given a pandas dataframe than anything else.

Reply:
I think the comment is just about the efficiency of the code, not about its behavior right @sarpu ?

so nothing to do with the input :)

Reply:
@lucasgautheron Yep, I just place my TODOs a bit too liberally